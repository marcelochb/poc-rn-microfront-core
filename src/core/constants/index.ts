/* istanbul ignore file */
export * from './core_constants';