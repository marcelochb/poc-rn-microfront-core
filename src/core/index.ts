/* istanbul ignore file */
export * from './constants';
export * from './services';