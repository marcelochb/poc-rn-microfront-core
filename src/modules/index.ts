
/* istanbul ignore file */
export * from './loan';
