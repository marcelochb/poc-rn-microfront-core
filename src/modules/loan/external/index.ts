/* istanbul ignore file */
export * from './datasources'