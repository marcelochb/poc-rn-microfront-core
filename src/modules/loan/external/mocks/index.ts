import { InternalAxiosRequestConfig } from "axios";
import { AxiosResponse } from "axios";
import { LoanEntity } from "../../domain";
import { LoanModel } from "../../infra";
export const loanModelList = [new LoanModel({id:'1', name: 'teste', type: 'tipo', amount: 'R$ 10'})];
export const loanModel = new LoanModel({id:'1', name: 'teste', type: 'tipo', amount: 'R$ 10'});
export const loanEntityList = [new LoanEntity({id:'1', name: 'teste', type: 'tipo', amount: 'R$ 10'})];
export const loanEntity = new LoanEntity({id:'1', name: 'teste', type: 'tipo', amount: 'R$ 10'});
export const getListMock:AxiosResponse<LoanModel[], any> = {
  data:[new LoanModel({id:'1', name: 'teste', type: 'tipo', amount: '10'})],
  status: 200,
  config: {} as InternalAxiosRequestConfig,
  headers: {},
  statusText: ''
}
export const getByMock:AxiosResponse<LoanModel, any> = {
  data:new LoanModel({id:'1', name: 'teste', type: 'tipo', amount: '10'}),
  status: 200,
  config: {} as InternalAxiosRequestConfig,
  headers: {},
  statusText: ''
}