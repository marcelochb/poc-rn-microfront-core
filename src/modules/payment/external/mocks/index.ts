import { PaymentEntity } from "../../domain";
import { PaymentModel } from "../../infra/models";
export const paymentEntity = new PaymentEntity({id:'1', recipient: 'teste', payer: 'payer', dueDate: '10/10/2023', amount: Number.parseFloat('10').toLocaleString("pt-BR",{minimumFractionDigits:2, style: 'currency', currency: 'BRL'})});
export const paymentEntityList = [paymentEntity];
export const paymentModel = new PaymentModel({id: '1', recipient: 'teste', payer: 'payer', dueDate: '10/10/2023', amount: Number.parseFloat('10').toLocaleString("pt-BR",{minimumFractionDigits:2, style: 'currency', currency: 'BRL'})});
export const paymentModelList = [paymentModel];
export const paymentJson = {id:'1', recipient: 'teste', payer: 'payer', dueDate: '10/10/2023', amount: '10'};
export const paymentJsonList = [paymentJson];