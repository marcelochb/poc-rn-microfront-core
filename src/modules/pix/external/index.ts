/* istanbul ignore file */
export * from './datasources';
export * from './mocks';