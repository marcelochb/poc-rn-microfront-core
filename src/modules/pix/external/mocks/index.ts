import { AxiosResponse, InternalAxiosRequestConfig } from "axios";
import { PixEntity } from "../../domain";
import { PixModel } from "../../infra/models";
export const pixEntity = new PixEntity({id:'1', name: 'teste', bank: 'bank', date: '10/10/2023', amount: Number.parseFloat('10').toLocaleString("pt-BR",{minimumFractionDigits:2, style: 'currency', currency: 'BRL'})});
export const pixEntityList = [pixEntity];
export const pixModel = new PixModel({id: '1', name: 'teste', bank: 'bank', date: '10/10/2023', amount: Number.parseFloat('10').toLocaleString("pt-BR",{minimumFractionDigits:2, style: 'currency', currency: 'BRL'})});
export const pixModelList = [pixModel];
export const pixJson = {id: '1', name: 'teste', bank: 'bank', date: '10/10/2023', amount: '10'};
export const pixJsonList = [pixJson];