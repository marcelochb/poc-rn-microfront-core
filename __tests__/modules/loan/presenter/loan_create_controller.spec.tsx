import 'reflect-metadata';
import { container } from 'tsyringe';
import { LoanConstants } from '../../../../src/core';
import { loanEntity } from '../../../../src/modules/loan/external/mocks';
import { useLoanCreateController, useLoanDetailController } from '../../../../src/modules';
import { Text, TouchableOpacity, View } from 'react-native';
import { fireEvent, render, screen, waitForElementToBeRemoved } from '@testing-library/react-native';
import React from 'react';




jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useRoute: jest.fn().mockImplementation(() => {return {params: {id:'1'}}} ),
}));
interface IProps {
  id: string;
}
describe('Loan Create Controller =>',() => {
  it('When success submit request should not return Error', async () => {
    class CreateLoanMockUsecase {
      call = jest.fn().mockImplementation( async () => null);
    }
    container.register(LoanConstants.CreateLoanUsecase,{useValue: new CreateLoanMockUsecase()});

    const MyComponent = () => {
      const {onSubmit, loading} = useLoanCreateController();
      return (
        <>
          <TouchableOpacity testID='subimit-button' onPress={() => onSubmit()}>Submit</TouchableOpacity>
          <Text>{loading ? 'loading':'success'}</Text>
        </>
        );
    }
    render(
        <MyComponent />
    )
    const button = screen.getByTestId('subimit-button');
    fireEvent.press(button);
    await waitForElementToBeRemoved(() => screen.getByText('loading'));
    expect(screen.toJSON()).toMatchSnapshot();        
    
  })
  it('When fail submit request should return Error', async () => {
    class CreateLoanMockUsecase {
      call = jest.fn().mockImplementation( async () => {throw Error});
    }
    container.register(LoanConstants.CreateLoanUsecase,{useValue: new CreateLoanMockUsecase()});

    const MyComponent = () => {
      const {onSubmit, loading, error} = useLoanCreateController();
      return (
        <>
          <TouchableOpacity testID='subimit-button' onPress={() => onSubmit()}>Submit</TouchableOpacity>
          <Text>{loading ? 'loading': error ? 'error':'success'}</Text>
        </>
        );
    }
    render(
        <MyComponent />
    )
    const button = screen.getByTestId('subimit-button');
    fireEvent.press(button);
    await waitForElementToBeRemoved(() => screen.getByText('loading'));
    expect(screen.toJSON()).toMatchSnapshot();        
    
  })

})