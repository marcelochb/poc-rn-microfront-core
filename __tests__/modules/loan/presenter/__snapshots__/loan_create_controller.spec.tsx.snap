// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loan Create Controller => When fail submit request should return Error 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="subimit-button"
  >
    Submit
  </View>,
  <Text>
    error
  </Text>,
]
`;

exports[`Loan Create Controller => When success submit request should not return Error 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="subimit-button"
  >
    Submit
  </View>,
  <Text>
    success
  </Text>,
]
`;
